controllers/blogController.js

```diff
@@ -0,0 +1,220 @@
+
+const Blog = require('../models/blogModel');
+const BlogDetails = require('../models/blogDetailModel');
+
+const getAllBlogs = async (req, res) => {
+
+    let offset = req.query.offset;
+    let limit = req.query.limit;
+
+    console.log(limit);
+
+    try {
+        const blogs = await Blog.find().skip(offset).limit(limit).populate('author', 'name email -_id');
+        res.status(200).json(
+            blogs
+        )
+    } catch (err) {
+        console.log(err);
+        res.status(500).json({
+            msg: "something went wrong, try again!!!",
+        })
+    }
+}
+
+const getPopularBlogs = async (req, res) => {
+
+    res.status(200).json({
+        msg: "get popular Blogs",
+    })
+}
+
+const getSimilarBlogs = async (req, res) => {
+    res.status(200).json({
+        msg: "get similar Blogs",
+    })
+}
+
+const getBlogById = async (req, res) => {
+    try {
+        const blogId = req.params.id;
+        const blog = await Blog.findById(blogId).populate('author', 'name email -_id');
+        res.status(200).json(
+            blog
+        )
+    } catch (err) {
+        console.log(err);
+        res.status(500).json({
+            msg: "something went wrong, try again!!!",
+        });
+    }
+}
+
+// draft/publish a blog
+const createBlog = async (req, res) => {
+
+    if (isObjectEmpty(req.body)) {
+        res.status(400).json({
+            msg: "Nothing to create. You provided all fields empty",
+        })
+        return;
+    }
+
+    if (!(req.body.title) || !(req.body.content)) {
+        res.status(400).json({
+            msg: "Please add both title and content",
+        })
+        return;
+    }
+
+    try {
+        // console.log(req.body.published); true/false/undefined
+        const newBlog = {
+            title: req.body.title,
+            author: req.user._id,
+            tags: req.body.tags,
+            short: req.body.short_desc === null ? "" : req.body.short_desc,
+            content: req.body.content,
+            published: (req.body.published ? req.body.published : false),
+        }
```

Comment:
we don't need to create an object like this. We can directly take the object from the req.body and assign it to Blog.create. This is because its the responsibility of frontend to send us the proper object.